components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -502,7 +517,14 @@ private void deleteDELETEOperationsTask(Timestamp timestamp) {
         }
         try {
             statement = connection.prepareStatement(sqlDeleteDELETETask);
-            statement.setLong(1, timestamp.getTime());
+
+            // create a nano time stamp relative to Unix Epoch
+            long currentStandardNano = timestamp.getTime() * 1000000;
+            long currentSystemNano = System.nanoTime();
+            currentStandardNano = currentStandardNano + (currentSystemNano - FrameworkServiceDataHolder.getInstance()
+                    .getNanoTimeReference());
+
+            statement.setLong(1, currentStandardNano);
```

Comment:
It is better if we can have a private method to calculate the currentStandardNano.